### DISCUSSION TITLE
Allowed characters in names

### DISCUSSION THREAD

**Discussion by lmoureaux** [hep-statistics-standard/hep-statistics-standard#24] [Standard]
As it stands, the spec allows for the following:
```json
"data": [
  {
    "name": "sin(a/b*2🐦)",
    "type": "point",
    "value": 0.0,
    "error": 1.0
  }
]
```
I let you guess how this would interact with `generic_function`. For less extreme example, some implementations might dislike names such as `if`, `function`, or `float` because they implement the parsing of `generic_function` using some scripting language.

I think we can go quite tight and restrict names to a letter or underscore followed by letters or numbers, only allowing the ASCII character set. I think implementations shouldn't be allowed to restrict this set further.

**Comment by cburgard** [comment]
I think this is a very reasonable proposal! I suggest we go with it until we hit a roadblock. Also, it's always easier to allow things that were forbidden in past versions than the other way around, so I suggest we start with a relatively tight restriction and loosen it when (justified) need arises.

**Reply by jonas-eschle** [reply]
Also agree, this sounds reasonable
